src/ppx_deriving_yojson_runtime.ml

```diff
@@ -8,7 +8,9 @@ let (>|=) x f =
 
 let rec map_bind f acc xs =
   match xs with
-  | x :: xs -> f x >>= fun x -> map_bind f (x :: acc) xs
+  | x :: xs -> (match f x with
+      | ((Result.Error _) as x) -> x
```

Comment:
I would use `e` or `err` instead of `x` here, which suggests a relation to the list elements.